packages/yew/src/html/conversion.rs

```diff
@@ -0,0 +1,89 @@
+use super::{Component, NodeRef, Scope};
+use std::{borrow::Cow, rc::Rc};
+
+/// Marker trait for types that the [`html!`] macro may clone implicitly.
+pub trait ImplicitClone: Clone {}
+
+// this is only implemented because there's no way to avoid cloning this value
+impl ImplicitClone for Cow<'static, str> {}
+
+impl<T: ImplicitClone> ImplicitClone for Option<T> {}
+impl<T> ImplicitClone for Rc<T> {}
+
+impl ImplicitClone for NodeRef {}
+impl<Comp: Component> ImplicitClone for Scope<Comp> {}
+// TODO there are still a few missing like AgentScope
+
+/// A trait similar to `Into<T>` which allows conversion to a value of a `Properties` struct.
+pub trait IntoPropValue<T> {
```

Comment:
Can we implement this for `T` and `&T` where `T: ToString`?

Reply:
No... I think it's because of specialization